tests/test_iotedgedev.py

```diff
@@ -1,266 +1,385 @@
+
+    assert "Directory is not empty" in result.output
 
-    os.chdir(test_solution_dir)
 
+def test_solution_create_in_empty_current_path(prepare_solution_with_env):
     dirname = "emptydir_current"
     os.makedirs(dirname)
     os.chdir(os.path.join(test_solution_dir, dirname))
 
-    cli = __import__("iotedgedev.cli", fromlist=['main'])
-    runner = CliRunner()
-    result = runner.invoke(cli.main, ['solution', 'new', '.'])
-    print(result.output)
+    result = runner_invoke(['solution', 'new', '.'])
 
     assert 'AZURE IOT EDGE SOLUTION CREATED' in result.output
 
 
-@pytest.fixture
-def test_solution_create_in_non_empty_dir(request):
-
+def test_solution_create_in_non_empty_dir(prepare_solution_with_env):
     os.chdir(tests_dir)
 
-    cli = __import__("iotedgedev.cli", fromlist=['main'])
-    runner = CliRunner()
-    result = runner.invoke(cli.main, ['solution', 'new', test_solution])
-    print(result.output)
+    result = runner_invoke(['solution', 'new', test_solution], True)
 
     assert "Directory is not empty" in result.output
 
 
-@pytest.fixture
-def test_solution_create_in_empty_child_dir(request):
-
-    os.chdir(test_solution_dir)
-
+def test_solution_create_in_empty_child_dir(prepare_solution_with_env):
     dirname = "emptydir"
     os.makedirs(dirname)
 
-    cli = __import__("iotedgedev.cli", fromlist=['main'])
-    runner = CliRunner()
-    result = runner.invoke(cli.main, ['solution', 'new', dirname])
-    print(result.output)
+    result = runner_invoke(['solution', 'new', dirname])
 
     assert 'AZURE IOT EDGE SOLUTION CREATED' in result.output
 
 
-def test_module_add():
-    """Test the addmodule command"""
-    os.chdir(test_solution_dir)
-    cli = __import__("iotedgedev.cli", fromlist=['main'])
-    runner = CliRunner()
-
-    add_module_and_verify(cli.main, runner, "c")
-    add_module_and_verify(cli.main, runner, "csharp")
-    add_module_and_verify(cli.main, runner, "java")
-
-    # Node.js modules is skipped on non-Windows for known issue.
-    # See more details: https://github.com/Azure/iotedgedev/issues/312
-    # See more details: https://github.com/Azure/iotedgedev/issues/346
-    if (platform.system().lower() == 'windows'):
-        add_module_and_verify(cli.main, runner, "nodejs")
-        launch_file = launch_json_file
-    else:
-        launch_file = launch_json_file_without_nodejs
-
-    add_module_and_verify(cli.main, runner, "python")
-    add_module_and_verify(cli.main, runner, "csharpfunction")
+def test_module_add(prepare_solution_with_env):
+    launch_file = launch_json_file
+    for template in templates:
+        # Node.js modules is skipped on non-Windows for below known issue.
+        # https://github.com/Azure/iotedgedev/issues/312
+        # https://github.com/Azure/iotedgedev/issues/346
+        if (template == "nodejs") and (platform.system().lower() != 'windows'):
+            launch_file = launch_json_file_without_nodejs
```

Comment:
You don't need to check `if (template == "nodejs") and (platform.system().lower() != 'windows')` in every iteration.